cartographer/mapping/internal/range_data_collator.cc

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer/mapping/internal/range_data_collator.h"
+
+#include <memory>
+
+#include "cartographer/common/make_unique.h"
+#include "cartographer/mapping/local_slam_result_data.h"
+#include "glog/logging.h"
+
+namespace cartographer {
+namespace mapping {
+
+sensor::TimedPointCloudOriginData RangeDataCollator::AddRangeData(
+    const std::string& sensor_id,
+    const sensor::TimedPointCloudData& timed_point_cloud_data) {
+  CHECK(expected_sensor_ids_.count(sensor_id) != 0);
```

Comment:
Is this CHECK really needed? Can this situation occur?

Reply:
Very unlikely to happen. But if it happens, we should really force-stop, so I would like to keep this check.